pkg/build/strategies/layered/layered.go

```diff
@@ -63,14 +80,25 @@ func (b *Layered) CreateDockerfile(config *api.Config) error {
 	}
 
 	buffer.WriteString(fmt.Sprintf("FROM %s\n", b.config.BuilderImage))
-	buffer.WriteString(fmt.Sprintf("COPY scripts %s\n", locations[0]))
+	// only COPY scripts dir if required scripts are present, i.e. the dir is not empty;
+	// even if the "scripts" dir exists, the COPY would fail if it was empty
+	scriptsIncluded := scriptsProvided(filepath.Join(config.WorkingDir, api.UploadScripts))
+	glog.V(2).Infof("Scripts dir has contents %v", scriptsIncluded)
```

Comment:
This will just return boolean to log, we can do better logging :-)

```go
if scriptsIncluded {
  glog.V(2).Infof("The scripts are included in %q directory", filepath.Join(config.WorkingDir, api.UploadScripts))
}
```

Also please use `path.Join()` here as this can run on Windows and we don't want to use Windows filepath separator.